pyi.py

```diff
@@ -2010,6 +2025,8 @@ def _visit_function(self, node: ast.FunctionDef | ast.AsyncFunctionDef) -> None:
 
         if self.in_class.active:
             self.check_self_typevars(node)
+        if self.in_protocol.active:
+            self.check_arg_kinds(node)
```

Comment:
minor micro-optimisation: we don't need to check whether we're in a protocol or not if we know we're not inside a class:

```suggestion
            if self.in_protocol.active:
                self.check_arg_kinds(node)
```